gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -238,6 +239,21 @@ private ManagedChannel createSingleChannel() throws IOException {
       // Will be overridden by user defined values if any.
       builder.keepAliveTime(DIRECT_PATH_KEEP_ALIVE_TIME_SECONDS, TimeUnit.SECONDS);
       builder.keepAliveTimeout(DIRECT_PATH_KEEP_ALIVE_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+      // When channel pooling is enabled, force the pick_first grpclb strategy.
+      // This is necessary to avoid the multiplicative effect of creating channel pool with
+      // `poolSize` number of  `ManagedChannel`s, each with a `subSetting` number of number of subchannels.
+      if (poolSize > 1) {
```

Comment:
Correct me if I'm wrong, for poolSize == 1, we still want to keep the 1-to-1 mapping between the ManagedChannel and the underlying subchannel, right? For example if in CloudPath the single managed channel is expected to have the stickiness to a same backend server, then for DirectPath we also need to keep this affinity.

Reply:
I think that's right. I updated the PR